ipie/addons/thermal/utils/ueg.py

```diff
@@ -0,0 +1,583 @@
+# Copyright 2022 The ipie Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Fionn Malone <fmalone@google.com>
+#          Joonho Lee
+#
+
+import numpy
+import scipy.sparse
+from ipie.utils.io import write_qmcpack_sparse
+
+# -----------------------------------------------------------------------------
```

Comment:
just `class UEG:` is fine. The # ---- docstrings don't add much here. Why is this in utils?

Reply:
The `ueg.py` was only meant for testing. I think the idea was to add a UEG example for users to refer to. I can add that.

Reply:
Right, if it's just an example I would put it in examples (not utils). If it's a bone-fide system it should go in Hamiltonian.

Reply:
It's called in `utils/testing.py` though--do we want it in the non-thermal `hamiltonian`? I thought it was discussed to only have `generic*.py` there at one point

Reply:
I think if we just want it for testing the more sensible (larger refactor would be), make a folder called testing, move all the testing methods into their own separate modules in testing/* and make a testing/hamiltonians.py to add new test only hamiltonians. Utils should disappear largely speaking #242. For now I guess this is ok.